File: idade.js
class CalculadoraIdade {
    constructor(){
        this.anoAtual = 0;
        this.anoNascimento = 0;
    }

    pedirAnos() {
        this.anoAtual = parseInt(prompt("Em que ano estamos?"));
        this.anoNascimento = parseInt(prompt("Em que ano você nasceu?"));
    }

    calcularIdade() {
        return this.anoAtual - this.anoNascimento;
    }

    mostrarIdade() {
        const idade = this.calcularIdade();
        console.log('Minha idade será ${idade}');
    }
}

//Uso
const calculadora = new calcularIdade();
calculadora.pedirAnos();
calculadora.mostrarIdade();

File: idade.rb
class CalculadoraIdade
    def initializa
        @ano_atual = 0
        @ano_nascimento = 0
    end

    def pedir_anos
        print "Em que ano estamos? "
        @ano_atual = gets.chomp.to_i
        print "Em que ano você nasceu? "
        @ano_nascimento = gets.chomp.to_i
    end

    def calcular_idade
        @ano_atual - @ano_nascimento
    end

    def mostrar_idade
        idade = calcular_idade
        puts "Minha idade será #{idade}"
    end
end